### PR TITLE
soc: arm: st_stm32: Move STM32H7_DUAL_CORE

### PR DESCRIPTION
**Pull request by Desvauxm-st**
Move STM32H7_DUAL_CORE to soc/arm/st_stm32/stm32h7/Kconfig.defconfig.series. Set default to y.
Cleanup occurences where was set to y.

### CODE REVIEW

File: soc/arm/st_stm32/stm32h7/Kconfig.defconfig.series [hunk 1]
```diff
@@ -17,4 +17,9 @@ config ROM_START_OFFSET
 	default 0x400 if BOOTLOADER_MCUBOOT
 	default 0x0   if !BOOTLOADER_MCUBOOT
 
+config STM32H7_DUAL_CORE
+	bool "Dual Core"
+	default y
```
Comment (erwango) [inline soc/arm/st_stm32/stm32h7/Kconfig.defconfig.series:22]: Please put it in `soc/arm/st_stm32/stm32h7/Kconfig.soc`

**Review by FRASTM — DISMISSED** [review]

File: soc/arm/st_stm32/stm32h7/Kconfig.soc [hunk 2]
```diff
@@ -77,4 +77,8 @@ config SOC_STM32H7B3XXQ
 	select CPU_CORTEX_M7
 	select CPU_HAS_FPU_DOUBLE_PRECISION
 
+config STM32H7_DUAL_CORE
+	bool "Dual Core"
+	default y
+
```
Comment (erwango) [inline soc/arm/st_stm32/stm32h7/Kconfig.soc:83]: It should be moved outside of choice structure.

**Review by erwango — CHANGES_REQUESTED** [review]

File: soc/arm/st_stm32/stm32h7/Kconfig.soc (hunk 2 above)
Comment (Desvauxm-st) [reply soc/arm/st_stm32/stm32h7/Kconfig.soc]: ok

**Review by FRASTM — COMMENTED** [review]
CI / merge conflicts to fix
Else LGTM

File: boards/arm/nucleo_h723zg/Kconfig.defconfig [hunk 3]
```diff
@@ -15,4 +15,7 @@ config NET_L2_ETHERNET
 
 endif # NETWORKING
 
+config STM32H7_DUAL_CORE
+	default n
```
Comment (erwango) [inline boards/arm/nucleo_h723zg/Kconfig.defconfig:19]: Not required. Kconfig symbols are `n` by default

File: boards/arm/nucleo_h743zi/Kconfig.defconfig [hunk 4]
```diff
@@ -15,4 +15,7 @@ config NET_L2_ETHERNET
 
 endif # NETWORKING
 
+config STM32H7_DUAL_CORE
+	default n
```
Comment (erwango) [inline boards/arm/nucleo_h743zi/Kconfig.defconfig:19]: Ditto, please remove

**Review by erwango — CHANGES_REQUESTED** [review]

File: soc/arm/st_stm32/stm32h7/Kconfig.soc [hunk 5]
```diff
@@ -78,3 +78,8 @@ config SOC_STM32H7B3XXQ
 	select CPU_HAS_FPU_DOUBLE_PRECISION
 
 endchoice
+
+config STM32H7_DUAL_CORE
+	bool "Dual Core"
+	depends on SOC_SERIES_STM32H7X
+	default y
```
Comment (erwango) [inline soc/arm/st_stm32/stm32h7/Kconfig.soc:85]: Don't default to `y`.
Use `select STM32H7_DUAL_CORE` in matching SoC definitins, like:
```
config SOC_STM32H747XX
	bool "STM32H747XX"
	select CPU_HAS_FPU_DOUBLE_PRECISION if CPU_CORTEX_M7
        select STM32H7_DUAL_CORE
```

(might require to move `STM32H7_DUAL_CORE`  definition before the choice section.

**Review by erwango — CHANGES_REQUESTED** [review]

File: soc/arm/st_stm32/stm32h7/Kconfig.soc (hunk 5 above)
Comment (Desvauxm-st) [reply soc/arm/st_stm32/stm32h7/Kconfig.soc]: ok

**Review by erwango — CHANGES_REQUESTED** [review]
Please update commit message which is not matching changesanymore.
Otherwise LGTM, thanks

**Review by erwango — APPROVED** [review]

**Review by gautierg-st — APPROVED** [review]

**Pull request merged**